app/controllers/comments_controller.rb

```diff
@@ -1,28 +1,45 @@
 class CommentsController < ApplicationController
+  before_action :article
   def create
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.create(comment_params)
-    redirect_to article_path(@article)
+    @comment = article.comments.new(comment_params) do |comment|
+      comment.user = current_user
+    end
+    if @comment.save
+      redirect_to article_path(article)
+    else 
+      redirect_to article_path(article)
+    end
   end
 
   def edit
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.edit(comment_params)
+    @comment = Comment.find(params[:id])
   end
 
   def update
     @comment = Comment.find(params[:id])
 
     if @comment.update(comment_params)
-      redirect_to article_path(@article)
+      redirect_to article_path(article)
     else
-      redirect_to article_path
+      render :edit
     end
   end
 
+  def destroy
+    @comment = Comment.find(params[:id])
+    @comment.destroy
+    redirect_to article_path(article)
+   
+
+  end
+
   private
 
   def comment_params
     params.require(:comment).permit(:commenter, :body)
   end
+
+  def article
+    @article = Article.find(params[:article_id])
```

Comment:
Let's improve this method. I have 2 ways to do it.

1. Remove this method from `before_action` and Change the code to 
`@article ||= Article.find(params[:article_id])`
then it will query article from DB  only first time and you can call article by `article.comments.new(comment_params)...`

2. Change this method name to `set_article`. Call it from `before_action` (as you did). Then call the article with `@article.comments.new(comment_params)...`